src/lib/coda_graphql/coda_graphql.ml

```diff
@@ -370,6 +370,39 @@ module Types = struct
               Stringable.Ledger_hash.to_base58_check
               @@ Staged_ledger_hash.ledger_hash staged_ledger_hash ) ] )
 
+  let natural_number_desc field_name num_bits =
+    Core.sprintf
+      !"String encoded natural number of the %s (%i bits)"
+      field_name num_bits
+
+  let consensus_state =
+    let open Consensus.Data.Consensus_state in
+    obj "ConsensusState" ~fields:(fun _ ->
+        [ field "blockchainLength" ~typ:(non_null string)
+            ~doc:
+              (natural_number_desc "blockchain length"
+                 Coda_numbers.Length.length_in_bits)
+            ~args:Arg.[]
+            ~resolve:(fun _ (t : Value.t) ->
+              Coda_numbers.Length.to_string @@ blockchain_length t )
+        ; field "slot"
+            ~doc:(natural_number_desc "slot in a block" 32)
+            ~typ:(non_null string)
+            ~args:Arg.[]
+            ~resolve:(fun _ t -> Unsigned.UInt32.to_string @@ curr_epoch t)
+        ; field "epoch"
+            ~doc:(natural_number_desc "epoch in a block" 32)
+            ~typ:(non_null string)
+            ~args:Arg.[]
+            ~resolve:(fun _ t -> Unsigned.UInt32.to_string @@ curr_epoch t)
+        ; field "totalCurreny"
+            ~doc:
+              (natural_number_desc "total currency in a block"
```

Comment:
This is `total currency` total in the whole network, not total in one block

Reply:
techically, it's the `total currency` of the whole network at the instance of the block